src/jdk.compiler/share/classes/com/sun/tools/javac/api/ClientCodeWrapper.java

```diff
@@ -195,13 +195,8 @@ TaskListener unwrap(TaskListener l) {
 
     protected boolean isTrusted(Object o) {
         Class<?> c = o.getClass();
-        Boolean trusted = trustedClasses.get(c);
-        if (trusted == null) {
-            trusted = c.getName().startsWith("com.sun.tools.javac.")
-                    || c.isAnnotationPresent(Trusted.class);
-            trustedClasses.put(c, trusted);
-        }
-        return trusted;
+        return trustedClasses.computeIfAbsent(c,
+                k -> c.getName().startsWith("com.sun.tools.javac.") || c.isAnnotationPresent(Trusted.class));
```

Comment:
```suggestion
                k -> k.getName().startsWith("com.sun.tools.javac.") || k.isAnnotationPresent(Trusted.class));
```
this generated lambda will be a singleton than created every call